src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+        }
+        else if (command.startsWith("event")) {
+            if (command.equals("event")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(5,6);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after event!");
+            }
+
+            int fromIndex = command.indexOf("/from");
+            int toIndex = command.indexOf("/to");
+
+            if (fromIndex == -1 && toIndex == -1) {
+                throw new TobiasException("Hey, please enter a /from and /to date/day/time!!");
+            }
+
+            if (fromIndex == -1) {
+                throw new TobiasException("Hey, please enter a /from date/day/time!!");
+            }
+
+            if (toIndex == -1) {
+                throw new TobiasException("Hey, please enter a /to date/day/time!!");
+            }
+
+            if (fromIndex > toIndex && toIndex > 0) {
+                throw new TobiasException("Hey, you have to enter /from before /to !!");
+            }
+
+            String from = command.substring(fromIndex + 6, toIndex);
+
+            if (from.isEmpty() || from.isBlank()) {
+                throw new TobiasException("Hey, please enter a /from date/day/time!!");
+            }
+
+            if ((toIndex + 4) > command.length()) {
+                throw new TobiasException("Hey, please enter a /to date/day/time!!");
+            }
+
+            String to = command.substring(toIndex + 4);
+
+            if (to.isEmpty() || to.isBlank()) {
+                throw new TobiasException("Hey, please enter a /to date/day/time!!");
+            }
+
+            String description = command.substring(6, fromIndex);
+
+            LocalDateTime f = dateFromString(from);
+            LocalDateTime t = dateFromString(to);
+
+            return new EventCommand(description, f, t);
+        }
+        else if (command.startsWith("mark")) {
+            int size = tasks.taskNum();
+
+            if (size == 0) {
+                throw new TobiasException("Your list is empty at the moment, add some todos/events/deadlines to MARK them!");
+            }
+
+            if (command.equals("mark")) {
+                throw new TobiasException("No index provided! Please give a valid index from 1 to " + size +" !!");
+            }
+
+            String blank = command.substring(4,5);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after mark!");
+            }
+
+            int index = Integer.parseInt(command.substring(5)) - 1;
+
+            if (index >= 0 && index < size) {
+                return new MarkCommand(index);
+            } else {
+                throw new TobiasException("Invalid number provided! Please give a valid index from 1 to " + size +" !!");
+            }
+        }
+        else if (command.startsWith("unmark")) {
```

Comment:
Maybe you would want to have else if statement on the same line as the closing brackets from the previous statement? This would follow our java coding standard:

```
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
```